src/rdkafka_conf.c

```diff
@@ -133,11 +133,11 @@ struct rd_kafka_property {
                 "OpenSSL >= 1.0.2 not available at build time"
 #endif
 
-#if OPENSSL_VERSION_NUMBER >= 0x10100000 && defined(WITH_SSL) && !defined(LIBRESSL_VERSION_NUMBER)
-#define _UNSUPPORTED_OPENSSL_1_1_0 .unsupported = NULL
+#if OPENSSL_VERSION_NUMBER >= 0x10100000 && defined(WITH_SSL) && !defined(OPENSSL_NO_ENGINE)
```

Comment:
Is it that engine-less OpenSSL builds are not exposing the ENGINE symbols that librdkafka uses, or that those API will fail?
My concern here is that when dynamically linked we can't really assume if engine support was built in or not.

Reply:
> Is it that engine-less OpenSSL builds are not exposing the ENGINE symbols that librdkafka uses, or that those API will fail? My concern here is that when dynamically linked we can't really assume if engine support was built in or not.

The former - configuring OpenSSL without any ENGINE support (`./config [...] no-engine`) defines `OPENSSL_NO_ENGINE`, so the ENGINE symbols aren't exposed (see https://github.com/openssl/openssl/blob/master/include/openssl/engine.h#L22: almost everything in that header file is gated by the `OPENSSL_NO_ENGINE` ifndef).

Reply:
Thanks for the clarification :+1:

Reply:
Does this potentially break builds using LibreSSL, since that define-check is removed?

Reply:
You're right, it does - LibreSSL identifies itself as OpenSSL 2.0.0, so that condition shouldn't have been removed. I'll re-add it.